Sources/_StringProcessing/Unicode/ASCII.swift

```diff
@@ -110,10 +113,10 @@ extension String {
     // Handle CR-LF:
     if base == ._carriageReturn && tail == ._lineFeed {
       utf8.formIndex(after: &next)
-      guard next == endIndex || utf8[next]._isSub300StartingByte else {
+      guard next == end || utf8[next]._isSub300StartingByte else {
         return nil
       }
-      assert(self[idx] == "\r\n")
+      // assert(self[idx] == "\r\n")
```

Comment:
What about these assertions?

Reply:
I was thinking this is the same case as above, but is it possible for a following scalar to combine with `"\n"`?

```swift
let s = "\r\n\u{301}"
let s1 = Substring(s.unicodeScalars.dropLast())
// s.count == 2
// s.first == "\r\n"
// s1.first == "\r\n"
```

Do we actually need the `_isSub300StartingByte` check here?